### ISSUE TITLE
NoSuchMethodError-Exception in net.dv8tion.jda.core.audio.AudioConnection, due to Java 9 changes

### ISSUE THREAD

**Issue by BBJip** [DV8FromTheWorld/JDA#540]
Hello,
i just updated on JDA 3.3.1_299 and noticed an interesting bug, by using JDA with Java 8. Due to the jigsaw changes in Java 9 the method `flip()` called on a buffer is no longer only implemented in the Buffer class itself. The method `java.nio.Buffer.flip()` is now overrided by the method `java.nio.ShortBuffer.flip()`. As a result of this and running the Java 9 compiled version JDA 3.3.1_299 with Java 8 a `NoSuchMethodError` Exception is thrown. I first noticed this issue by using the AudioSendHandler interface.

```
[30.10.2017] [16:12:32] [ERROR] [DefaultSendSystem] Encountered an Exception 
 java.lang.NoSuchMethodError: java.nio.ShortBuffer.flip()Ljava/nio/ShortBuffer;
	at net.dv8tion.jda.core.audio.AudioConnection.encodeToOpus(AudioConnection.java:537) ~[JDA-3.3.1_299.jar:3.3.1_299]
	at net.dv8tion.jda.core.audio.AudioConnection.access$900(AudioConnection.java:54) ~[JDA-3.3.1_299.jar:3.3.1_299]
	at net.dv8tion.jda.core.audio.AudioConnection$PacketProvider.getNextPacket(AudioConnection.java:620) ~[JDA-3.3.1_299.jar:3.3.1_299]
	at net.dv8tion.jda.core.audio.factory.DefaultSendSystem.lambda$start$0(DefaultSendSystem.java:57) ~[JDA-3.3.1_299.jar:3.3.1_299]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_144]
```

**Comment by MinnDevelopment** [comment]
I am unable to reproduce this.

**Comment by kantenkugel** [comment]
can you give us more infos?
- What operating system are you using?
- What JRE is your bot executed with (`java -version`)?
- How/where did you download JDA and its dependencies (maven/gradle/manual [from where])?

**Comment by BBJip** [comment]
My apologizes for the less provided informations, i thougt that my description were sufficient enough.
**Windows 10 Version 1709.
Project SDK is 1.8.0_144
Project Language Level: 8 (important for reproducing the erro)
My maven pom:**
```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <groupId>de.blackbrownies</groupId>
    <artifactId>jdatest</artifactId>
    <version>1.0-SNAPSHOT</version>

    <repositories>
        <repository>
            <id>jcenter</id>
            <name>jcenter-bintray</name>
            <url>http://jcenter.bintray.com</url>
        </repository>
    </repositories>
    <dependencies>
        <dependency>
            <groupId>net.dv8tion</groupId>
            <artifactId>JDA</artifactId>
            <version>3.3.1_299</version>
        </dependency>
    </dependencies>

</project>
```
**My Main class:** 
```
import net.dv8tion.jda.core.AccountType;
import net.dv8tion.jda.core.JDA;
import net.dv8tion.jda.core.JDABuilder;
import net.dv8tion.jda.core.OnlineStatus;
import net.dv8tion.jda.core.audio.AudioSendHandler;
import net.dv8tion.jda.core.exceptions.RateLimitedException;

import javax.security.auth.login.LoginException;

public class Main {
    public static void main(String[] args) {
        JDABuilder builder = new JDABuilder(AccountType.BOT);
        builder.setToken("MzYwMDg1MDEyNjUxMTgwMDMy.DKQiLg.ssbHhF44cvBfD8iq6KbHpL3WNAU");
        builder.setAutoReconnect(true);
        builder.setStatus(OnlineStatus.ONLINE);
        JDA jda = null;
        try {
            jda = builder.buildBlocking();
            jda.getGuilds().get(0).getAudioManager().setSendingHandler(new AudioSendHandler() {
                public boolean canProvide() {
                    return true;
                }

                public byte[] provide20MsAudio() {
                    return new byte[3840];
                }
            });
            jda.getGuilds().get(0).getAudioManager().openAudioConnection(jda.getGuilds().get(0).getVoiceChannels().get(0));
        } catch (LoginException e) {
            e.printStackTrace();
        } catch (InterruptedException e) {
            e.printStackTrace();
        } catch (RateLimitedException e) {
            e.printStackTrace();
        }
    }
}
```

My presumption, as described earlier, is that JDA 3.3.1_299 is compiled with java 9, where the method `java.nio.Buffer.flip()` is overwritten by `java.nio.ShortBuffer.flip()` (Used at `net.dv8tion.jda.core.audio.AudioConnection.encodeToOpus(AudioConnection.java:537`). If i use Java 1.8.0_144 and Project Language Level 8, then i get a `java.lang.NoSuchMethodError` Exception, because my runtime tries to call the method `java.nio.ShortBuffer.flip()`, which does not exist under java 8 and should be `java.nio.Buffer.flip()`.

**Comment by kantenkugel** [comment]
Just wanted to edit and inform you that you should revoke the token :)
Would still regenerate it... (as the history of issues is visible)

**Comment by kantenkugel** [comment]
Can verify bug with java 1.8_121 and his code

**Comment by BBJip** [comment]
@kantenkugel yeah, thanks i did completle forgot the token xD

**Comment by kantenkugel** [comment]
Fixed by 9996b3add4f475097f3db151162757d26ada7624 (build 300)
